src/components/Checkbox/Checkbox.tsx

```diff
@@ -87,9 +87,14 @@ export const CheckboxContainer = styled.div`
 `;
 
 export const BaseIcon = styled(Icon)`
-  overflow: visible;
-  height: 1em;
-  width: 1em;
+  ${({ color }) => {
+    return `
+      overflow: visible;
+      color: ${color};
+      width: 1em;
+      height: 1em;
+    `;
+  }}
```

Comment:
I think this can be simplified a bit

```suggestion
  ${({ color }) => `
    overflow: visible;
    color: ${color};
    width: 1em;
    height: 1em;
  `}
```

Reply:
a bit more?
```typescript
    overflow: visible;
    color: ${({ color }) => color};
    width: 1em;
    height: 1em;
```

Reply:
Gets a bit weird when you add types to the props if you're using a type definition - it ends up being a Pick, which gets kinda awkward. That's the main reason I like having them all imported at the top

Reply:
Pretty sure the styled function is generic, i.e.

```js
export const MyComp = styled.div<{tall: boolean; color: string}>`
  height: ${({ tall }) => tall ? 2 : 1}rem;
  color: ${({ color }) => color};
`;
```

Reply:
or maybe it gets weird because of the `as StyledSubcomponentType` cast?